File: kotlin/PlanesAndroid/app/src/test/java/GuessPointTest.java
import com.google.common.truth.Truth;
import org.junit.Test;
import static com.google.common.truth.Truth.assertThat;

import com.planes.common.GuessPoint;
import com.planes.common.Type;


public class GuessPointTest {
    @Test
    public void GuessPoint_SetType() {
        GuessPoint gp1 = new GuessPoint(1, 1);
        GuessPoint gp2 = new GuessPoint(1, 1, Type.Dead);
        gp1.setType(Type.Dead);
        assertThat(gp1.equals(gp2)).isTrue();
    }

    @Test
    public void GuessPoint_Clone() {
        GuessPoint gp1 = new GuessPoint(1, 1, Type.Dead);
        GuessPoint gp2 = (GuessPoint)gp1.clone();
        assertThat(gp1.equals(gp2)).isTrue();
    }
}

File: kotlin/PlanesAndroid/app/src/test/java/GuessPointTest.kt
import com.google.common.truth.Truth
import com.planes.single_player_engine.GuessPoint
import com.planes.single_player_engine.Type
import org.junit.Test

class GuessPointTest {
    @Test
    fun GuessPoint_SetType() {
        val gp1 = GuessPoint(1, 1)
        val gp2 = GuessPoint(1, 1, Type.Dead)
        gp1.setType(Type.Dead)
        Truth.assertThat(gp1.equals(gp2)).isTrue()
    }

    @Test
    fun GuessPoint_Clone() {
        val gp1 = GuessPoint(1, 1, Type.Dead)
        val gp2 = gp1.clone() as GuessPoint
        Truth.assertThat(gp1.equals(gp2)).isTrue()
    }
}